django-backend/fecfiler/transactions/utils.py

```diff
@@ -37,3 +38,48 @@ def add_location(data, contact: Contact, identifier: str):
     data[f"{identifier}_city"] = contact.city
     data[f"{identifier}_state"] = contact.state
     data[f"{identifier}_zip"] = contact.zip
+
+
+def filter_for_previous_transactions(
+    queryset,
+    transaction_id,
+    date,
+    aggregation_group,
+    contact_1_id=None,
+    contact_2_id=None,
+    election_code=None,
+    election_year=None,
+    office=None,
+    state=None,
+    district=None,
+):
+    query = queryset.filter(
+        ~Q(id=transaction_id or None),
+        Q(date__year=date.year),
+        Q(date__lte=date),
+        Q(aggregation_group=aggregation_group),
+    )
+    if contact_1_id:
+        query = query.filter(Q(contact_1_id=contact_1_id))
+    elif contact_2_id:
+        query = query.filter(
+            Q(contact_2_id=contact_2_id),
+            Q(schedule_f__general_election_year=election_year)
+        )
+    else:
+        query = query.filter(
+            Q(schedule_e__election_code=election_code),
+            Q(contact_2__candidate_office=office),
+            Q(contact_2__candidate_state=state),
+            Q(contact_2__candidate_district=district),
+        )
+
+    original_transaction = None
+    if transaction_id:
+        original_transaction = queryset.get(id=transaction_id)
+        query = query.filter(
+            Q(created__lt=original_transaction.created) | ~Q(date=date)
```

Comment:
should this be ~Q(date >= date) to make sure we're always getting previous transactions? Also, couldn't transactions with older created dates still have newer dates?

Reply:
I'm pretty convinced that I've got all the logic working correctly, and I think I have a pretty comprehensive set of unit and E2E tests, but if you see even a chance that there's room for an edge-case, then we should definitely take that seriously.  I'll take another look at my unit tests and make sure that there are tests for the aggregation logic around creation order

Reply:
I think it's correct @Elaine-Krauss-TCG maybe we can just add a comment explaining the purpose of this one filter.  I think the confusion is in thinking it's meant to be the whole filter, when really it's just addressing a subset of the transactions that are filtered in the whole function (ones that are the same date but created before the original)

Reply:
yes @toddlees I think I misunderstood this filter - thanks for all the checking and clarification @toddlees @Elaine-Krauss-TCG